apps/api/src/main.ts

```diff
@@ -1,97 +1,49 @@
 import { NestFactory } from '@nestjs/core';
-import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';
 import { AppModule } from './app/app.module';
 import { ConfigService } from '@nestjs/config';
-import { Logger, RequestMethod, ValidationPipe } from '@nestjs/common';
-import config from './configs/config';
+import {
+  INestApplication,
+  Logger,
+  RequestMethod,
+  ValidationPipe,
+  VersioningType,
+} from '@nestjs/common';
 import helmet from 'helmet';
-import { BackendConfig } from './configs/backend.interface';
-import { HelmetConfig } from './configs/helmet.interface';
-import { CorsConfig } from './configs/cors.interface';
-import { SwaggerConfig } from './configs/swagger.interface';
-import * as basicAuth from 'express-basic-auth';
+import { ENV_NAME } from '@musica/core';
+import { setupSwagger } from '@musica/core';
 
 async function bootstrap() {
-  const app = await NestFactory.create(AppModule);
+  const app: INestApplication = await NestFactory.create(AppModule);
 
   // Validation
   app.useGlobalPipes(new ValidationPipe());
 
-  // Retrieve configuration values using configService
   const configService = app.get(ConfigService);
+  const port = configService.get<number>(ENV_NAME.PORT_API);
 
-  const backendConfig: BackendConfig = {
-    port: configService.get<number>('PORT_API') || config.backend.port,
-  };
-
-  const helmetConfig: HelmetConfig = {
-    enabled:
-      configService.get<boolean>('HELMET_ENABLED') ?? config.helmet.enabled,
-  };
-
-  const swaggerConfig: SwaggerConfig = {
-    enabled:
-      configService.get<boolean>('SWAGGER_ENABLED') ?? config.swagger.enabled,
-    title: configService.get<string>('SWAGGER_TITLE') ?? config.swagger.title,
-    description:
-      configService.get<string>('SWAGGER_DESC') ?? config.swagger.description,
-    version: configService.get<string>('API_VERSION') ?? config.swagger.version,
-    path: configService.get<string>('SWAGGER_PATH') ?? config.swagger.path,
-    user: configService.get<string>('SWAGGER_USER') ?? config.swagger.user,
-    password:
-      configService.get<string>('SWAGGER_PASSWORD') ?? config.swagger.password,
-    env: configService.get<string[]>('SWAGGER_ENV') ?? config.swagger.env,
-  };
-  const corsConfig: CorsConfig = {
-    enabled: configService.get<boolean>('CORS_ENABLED') ?? config.cors.enabled,
-  };
-
-  if (corsConfig.enabled) {
+  if (configService.get<boolean>(ENV_NAME.CORS_ENABLED)) {
     app.enableCors();
   }
 
-  if (helmetConfig.enabled) {
+  if (configService.get<boolean>(ENV_NAME.HELMET_ENABLED)) {
     app.use(helmet());
   }
 
   // set api version
-  app.setGlobalPrefix('api/' + 'v' + swaggerConfig.version, {
+  app.enableVersioning({
+    type: VersioningType.URI,
+  });
```

Comment:
1. extract method and remove comment
2. use template_string 
3. use a better variable name rather than `v`

## reference: 
* https://refactoring.guru/extract-method

Reply:
Also, we could use content-type instead of versioning api via path

Reply:
https://medium.com/@anayooleru/creating-a-version-control-strategy-for-your-api-b2ee62b3189e#:~:text=Versioning%20with%20Content%2DType&text=A%20custom%20Content%20type%20is,can%20be%20in%20different%20versions.